vizro-core/src/vizro/models/_page.py

```diff
@@ -214,9 +206,28 @@ def build(self) -> _PageBuildType:
         components_container = _build_inner_layout(self.layout, self.components)
         components_container.id = "page-components"
 
+        # Components that are required to make action chains function correctly:
+        #   - {action.id}_guarded_trigger for the first action in a chain so that guard_action_chain can prevent
+        #     undesired triggering (workaround for Dash prevent_initial_call=True behaviour)
+        #   - {action.id}_finished for completion of an action callback to trigger the next action in the chain
+        #   - action._dash_components for particular actions (e.g. dcc.Download for export_data) - hopefully will be
+        #     removed in future
+        # These components are recreated on every page rather than going at the global dashboard level so that we do
+        # not accidentally trigger callbacks (workaround for Dash prevent_initial_call=True behaviour).
+        action_components = []
```

Comment:
Key change: these components are what's needed for actions chains to work. Previously this sort of thing was done in `ActionsLoop`.